scripts/cve/cve_due_dates.py

```
@@ -0,0 +1,460 @@
+                timeout=30
+            )
+            response.raise_for_status()
+            if response.text:
+                return response.json()
+            return {}
+        else:
+            if params:
+                query_string = "&".join(f"{k}={urllib.parse.quote(str(v))}" for k, v in params.items())
+                url = f"{url}?{query_string}"
+
+            req = urllib.request.Request(url, headers=self.headers, method=method)
```

Ruff (0.15.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)